src/toggle/toggle.tsx

```diff
@@ -0,0 +1,163 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import {
+    Container,
+    Input,
+    Label,
+    // SubLabel,
+    TextContainer,
+} from "./toggle.styles";
+import { ToggleProps } from "./types";
+import { ToggleIcon, ToggleIconType } from "../shared/toggle-icon/toggle-icon";
+
+export const Toggle = ({
+    type = "checkbox",
+    indicator,
+    checked,
+    styleType = "default",
+    children,
+    // subLabel,
+    disabled,
+    error,
+    name,
+    id,
+    className,
+    "data-testid": testId,
+    onChange,
+}: ToggleProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const [selected, setSelected] = useState<boolean>(checked);
+    const [inputId] = useState<string>(
+        id ? `${id}-input` : crypto.randomUUID()
+    );
+    const inputRef = useRef<HTMLInputElement>();
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        setSelected(checked);
+    }, [checked]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleOnChange = useCallback(
+        (event: React.ChangeEvent<HTMLInputElement>) => {
+            if (!disabled) {
+                switch (type) {
+                    case "checkbox":
+                        {
+                            if (!selected) {
+                                setSelected((prevSelected) => {
+                                    return !prevSelected;
+                                });
+                            }
+                        }
+                        break;
+                    case "radio":
+                    case "yes":
+                    case "no":
+                        {
+                            if (!selected) {
+                                setSelected(true);
+                            }
+                        }
+                        break;
+                }
+
+                if (onChange) onChange(event);
+            }
+        },
+        []
```

Comment:
should include `onChange` dependency here

Reply:
actually i dont think this `useCallback` is required. Will remove.